linera-base/src/identifiers.rs

```diff
@@ -574,6 +584,83 @@ impl std::str::FromStr for Owner {
     }
 }
 
+impl Serialize for AccountOwner {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            AccountOwner::User(owner) => {
+                let key = format!("User:{}", owner);
+                serializer.serialize_str(&key)
+            }
+            AccountOwner::Application(app_id) => {
+                let key = format!("Application:{}", app_id);
+                serializer.serialize_str(&key)
+            }
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for AccountOwner {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct AccountOwnerVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for AccountOwnerVisitor {
+            type Value = AccountOwner;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a string representing an AccountOwner")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                let parts = value.splitn(2, ':').collect::<Vec<_>>();
+                if parts.len() != 2 {
+                    return Err(Error::custom("string does not contain colon"));
+                }
+
+                match parts[0] {
+                    "User" => {
+                        let owner = Owner::from_str(parts[1]).map_err(|_| {
+                            Error::custom(format!(
+                                "failed to parse Owner from string: {}",
+                                parts[1]
+                            ))
+                        })?;
+                        Ok(AccountOwner::User(owner))
+                    }
+                    "Application" => {
+                        let app_id = ApplicationId::from_str(parts[1]).map_err(|_| {
+                            Error::custom(format!(
+                                "failed to parse ApplicationId from string: {}",
+                                parts[1]
+                            ))
+                        })?;
+                        Ok(AccountOwner::Application(app_id))
+                    }
+                    _ => Err(Error::unknown_variant(parts[0], &["User", "Application"])),
+                }
+            }
+        }
+        deserializer.deserialize_str(AccountOwnerVisitor)
+    }
+}
```

Comment:
This could be much more concise with an implementation like the one for `PublicKey` or `Amount`, without manually implementing a visitor.

(I know you only moved this code; but since it's going to be used in more contexts now, it might be good to improve it.)